src/main/java/duke/DukeList.java

```diff
@@ -0,0 +1,147 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+
+/**
+ * DukeList stores tasks given by users to the main Dukebot.
+ */
+public class DukeList {
+    //The ArrayList used to store all tasks accordingly.
+    private ArrayList<Task> tasks;
+    //The path directory to test.s
+    private String path;
+    //The DataSaver class used to transmit and store information during the use of the Duke class
+    private DataSaver middleman;
+
+    /**
+     * Instantiates a new DukeList.
+     * @param path path directory for the test.s file.
+     */
+    public DukeList(String path) {
+        this.path = path;
+        try {
+            this.middleman = new DataSaver(path);
+            this.tasks = middleman.load();
+        } catch (IOException | ClassNotFoundException e) {
+            UI.errorMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Saves the data stored in the DukeList into the data.s file.
+     */
+    public void save() {
+        middleman.store(this.tasks);
+    }
+
+
+
+    /**
+     * Helps to store user todo tasks into the list.
+     *
+     * @param task takes in the task given by the user.
+     */
+    public void store(String task) {
+        Task todo = new ToDo(task);
+        tasks.add(todo);
+        int index = tasks.size();
+        UI.store(task, index);
+    }
+
+    /**
+     * Helps to store user deadline tasks and deadlines into the list.
+     *
+     * @param task takes in the task given by the user.
+     * @param deadline takes in the deadline of the deadline task.
+     */
+    public void store(String task, String deadline) {
+        Task todo = new Deadline(task, deadline);
```

Comment:
I think it could get confusing if you name the Task as todo, when it's specifically a Deadline and not a ToDo. Perhaps you could consider renaming it to deadline? Also applies to the Event in the store method below (line 74).